lib/jxl/base/status.h

```diff
@@ -441,19 +357,20 @@ class JXL_MUST_USE_RESULT StatusOr {
 #define PRIVATE_JXL_ASSIGN_OR_RETURN_IMPL(name, lhs, statusor) \
   auto name = statusor;                                        \
   JXL_RETURN_IF_ERROR(name.status());                          \
-  lhs = std::move(name).value();
+  lhs = std::move(name).value_();
 // NOLINTEND(bugprone-macro-parentheses)
 
-// NB: do not use outside of tests / tools!!!
-#define JXL_ASSIGN_OR_DIE(lhs, statusor) \
-  PRIVATE_JXL_ASSIGN_OR_DIE_IMPL(        \
-      JXL_JOIN(assign_or_die_temporary_variable, __LINE__), lhs, statusor)
+#define JXL_ASSIGN_OR(lhs, statusor, code)                                     \
```

Comment:
I don't really like passing arbitrary code here, how do we use this now?

Reply:
Replaced with more straightforward `JXL_ASSIGN_OR_QUIT`. 3-4 cases where this does not match reverted to `JXL_ASSIGN_OR_RETURN` and wrapped in a closure.